source/common/upstream/cluster_manager_impl.cc

```diff
@@ -1348,6 +1348,31 @@ ClusterManagerImpl::ThreadLocalClusterManagerImpl::ClusterEntry::ClusterEntry(
                                                      parent.parent_.random_, cluster->lbConfig());
       break;
     }
+    case LoadBalancerType::LoadBalancingPolicyConfig: {
+      ASSERT(lb_factory_ == nullptr);
+      for (const auto& policy : cluster->loadBalancingPolicy().policies()) {
+        LoadBalancerFactoryContextImpl context(
+            parent_.parent_.validation_context_.staticValidationVisitor());
+        TypedLoadBalancerFactory* factory =
+            Registry::FactoryRegistry<TypedLoadBalancerFactory>::getFactory(policy.name());
+
+        if (factory == nullptr) {
+          ENVOY_LOG(warn, fmt::format("Didn't find a registered implementation for name: '{}'",
+                                      policy.name()));
+          continue;
```

Comment:
Once we fix the above comment you can use the throwing version of the factory lookup functions.